File: e2e-pytest-playwright/pages/payment_page.py
from pages.abstract_page import AbstractPage


class PaymentPage(AbstractPage):

  def enter_holder_name(self, holder_name):
    self.driver.locator("[name='holder_name']").fill(holder_name)

  def enter_card_number(self, card_number):
    self.driver.locator("[name='card_number']").fill(card_number)

  def enter_expiry_month(self, expiry_month):
    elem = self.driver.locator("[name='expiry_month']")
    elem.select_option(expiry_month)

  def enter_expiry_year(self, expiry_year):
    elem = self.driver.locator("[name='expiry_year']")
    elem.select_option(expiry_year)

  def click_pay_now(self):
    self.driver.locator("//input[@value='Pay now']").click()
 
  def select_card_type(self, card_type):
    self.driver.locator("//input[@name='card_type' and @value='" + card_type + "']").click()
  

File: e2e-mocha-selenium/pages/payment_page.js
const AbstractPage = require('./abstract_page');

var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;

class PaymentPage extends AbstractPage {

  constructor(driver) {
    super(driver);
  }

  async selectVisa() {
    await this.driver.findElement(By.xpath("//input[@name='card_type' and @value='visa']")).click();
  }
  
  async enterHolderName(name) {
    await this.driver.findElement(By.name("holder_name")).sendKeys(name);
  }

  async enterCardNumber(card_no) {
    await this.driver.findElement(By.name("card_number")).sendKeys(card_no);
  }

   
  async selectExpiryMonth(month) {
    await this.driver.findElement(By.name("expiry_month")).sendKeys(month);
  }

  async selectExpiryYear(year) {
    await this.driver.findElement(By.name("expiry_year")).sendKeys(year);
  }

  async clickPayNow() {
    await this.driver.findElement(By.xpath("//input[@value='Pay now']")).click()
  }

};

module.exports = PaymentPage;
